MonoGame.Framework/Content/ContentManager.cs

```diff
@@ -372,18 +374,83 @@ public virtual T Load<T>(string assetName)
 
             // Load the asset.
             result = ReadAsset<T>(assetName, null);
-
             loadedAssets[key] = result;
+
             return result;
 		}
 
+        /// <summary />
+		protected T ReadAsset<T>(string assetName, Action<IDisposable> recordDisposableObject)
+		{
+			if (string.IsNullOrEmpty(assetName))
+			{
+				throw new ArgumentNullException("assetName");
+			}
+			if (disposed)
+			{
+				throw new ObjectDisposedException("ContentManager");
+			}
+
+			string originalAssetName = assetName;
+			object result = null;
+
+            try
+            {
+                // Try to load as XNB file
+                var stream = OpenStream(assetName);
+                using (var xnbReader = new BinaryReader(stream))
+                {
+                    using (var reader = GetContentReaderFromXnb(assetName, stream, xnbReader, recordDisposableObject))
+                    {
+                        result = reader.ReadAsset<T>();
+                        if (result is GraphicsResource)
+                        {
+                            ((GraphicsResource)result).Name = originalAssetName;
+                        }
+                    }
+                }
+            }
+            catch (ContentLoadException)
```

Comment:
We definitely need some Unit tests as this "should" be a "Stream open Exception" and not a "ContentLoadException" which was a custom exception the framework originally threw but has been moved to line 441.

Reply:
OpenStream doesn't throw anything but ContentLoadException since every kind of exception thrown in its body is catched (line 484), and ContentLoadException are thrown in other exception paths as well.